### ISSUE TITLE
False positive for commented code

### ISSUE THREAD

**Issue by TCABJ** [jimhester/lintr#451]
It's possible to generate a false positive for commented code by creating an object with a name containing two hashtags separated by any alpha-numeric character. 

Linted code:
```R
c("#a#" = 1)
```

Message:
```
.../test.R style: Commented code should be removed.
c("#a#" = 1)
    ^~~~~~~~
```

This should not be flagged as commented code.

**Comment by AshesITR** [comment]
The fix I did does open up the edge case
```
c("#a#" = 1) # the <- actual_commented_code
```
